Assets/_BForBoss/_Utility/Scripts/ObjectPooler.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using UnityEngine;
+#if UNITY_2021_1_OR_NEWER
+using UnityEngine.Pool;
+#else
+using UnityEngine.SceneManagement;
+using System.Collections.Generic;
+#endif
+
+namespace Perigon.Utility
+{
+    public class ObjectPooler<T> where T : Component
+    {
+		protected string name;
+
+#if UNITY_2021_1_OR_NEWER
+		protected IObjectPool<T> objectPool;
+#else
+		protected Scene poolScene;
+		protected List<T> pool = null;
+#endif
+
+		protected Func<T> createFunc;
+		protected Action<T> actionOnGet;
+		protected Action<T> actionOnRelease;
+
+        public ObjectPooler(string poolName, Func<T> createFunc, Action<T> actionOnGet, Action<T> actionOnRelease)
```

Comment:
Are these Actions and Functions Nullable?
and what is 
`createFunc` means?

Reply:
Oh okay, so `creatFunc` is like a builder. 
What do you think of the arg name - `objectBuilder`

Reply:
Sure I can make that change, I think an xml comment on the constructor is probably a good idea too

Reply:
As long as Unity doesn't change their ObjectPool constructor, that should be okay.

Reply:
And yeah good catch, the actions are nullable but the function isn't, I'll make that explicit